android-components/components/feature/media/src/main/java/mozilla/components/feature/media/service/AbstractMediaSessionService.kt

```diff
@@ -35,6 +36,7 @@ internal class MediaServiceBinder(delegate: MediaSessionDelegate) : Binder() {
  */
 abstract class AbstractMediaSessionService : Service() {
     protected abstract val store: BrowserStore
+    protected abstract val crashReporter: CrashReporting
```

Comment:
We could avoid a breaking change by making this property optional and using `null` as a default through the instantiation chain.

Reply:
I am not necessarily suggesting that we _should_ not introduce a breaking change here but am curious whether this crash reporting is something other downstreams will want as a hard requirement vs only needing it in Fenix.

Reply:
This is kind of what I was thinking with my first approach using a `Fact`. I think if we're going to go the route of using `CrashReporting` the opt-out should be explicit using a no-op `CrashReporting` object versus requiring the domain knowledge of opting in.

Reply:
I was thinking more generally from the point of view of controlling dependencies and exposing the API. If we're injecting something into a component the caller should be responsible for deciding what to do, not the component.

Reply:
Sorry, I have a boring response: it depends.

It depends on what the thing is. My weak opinion is that having the `CrashReporting` null by default is nicer because, as an embedder of the media component, I don't want to implement crash reporting in order to use the feature (it's not part of the core functionality).

We have [examples](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/AbstractCustomTabsService.kt#42) where there is non-core functionality of a service that is null by default because of non-core functionality. 

Again, it depends. 🙂

Reply:
I'm good with this :)